lib/ts/recipe/emailverification/components/features/emailVerification/index.tsx

```diff
@@ -87,7 +88,24 @@ export const EmailVerification: React.FC<Prop> = (props) => {
         },
         [props.recipe, setStatus, onSuccess]
     );
-    useOnMountAPICall(fetchIsEmailVerified, checkIsEmailVerified, undefined, sessionContext.loading === false);
+
+    const redirectToAuthWithHistory = useCallback(async () => {
+        await redirectToAuth({ redirectBack: false, history: props.history });
+    }, [props.history]);
+
+    const handleError = useCallback(
+        async (err) => {
+            // If the error cleared the session we redirect away, otherwise we have no way of handling it.
+            if (await Session.getInstanceOrThrow().doesSessionExist({ userContext })) {
+                throw err;
+            } else {
+                await redirectToAuthWithHistory();
+            }
+        },
+        [redirectToAuthWithHistory]
+    );
```

Comment:
Add a TODO stating that when we add SessionAuth with requireAuth=true, we should remove this.